melkor/models/sklearn_pipeline.py

```diff
@@ -0,0 +1,110 @@
+import sklearn
+import pandas as pd
+import numpy as np
+import pickle
+from nptyping import NDArray, Float32
+from typing import Union
+from sklearn.pipeline import Pipeline
+from sklearn.compose import ColumnTransformer
+
+# TODO: Add scoring metrics
+# TODO: Change y to 1d array
+
+
+class SKLearnModelPipeline:
+    def __init__(
+        self,
+        config: dict = {},
+        cat_cols: list = [],
+        num_cols: list = [],
+        inference: bool = False,
+    ) -> None:
+        self.config = config
+        self.cat_cols = cat_cols
+        self.num_cols = num_cols
+        if not inference:
```

Comment:
Why not doing a generic Pipeline object with the basic functionalities and then split the child objects into train/inference.